mod/state-transition/pkg/core/state_processor_genesis.go

```diff
@@ -44,126 +44,127 @@ func (sp *StateProcessor[
 ]) InitializePreminedBeaconStateFromEth1(
 	st BeaconStateT,
 	deposits []DepositT,
-	executionPayloadHeader ExecutionPayloadHeaderT,
+	execPayloadHeader ExecutionPayloadHeaderT,
 	genesisVersion common.Version,
 ) (transition.ValidatorUpdates, error) {
 	sp.processingGenesis = true
 	defer func() {
 		sp.processingGenesis = false
 	}()
 
-	var (
-		blkHeader BeaconBlockHeaderT
-		blkBody   BeaconBlockBodyT
-		fork      ForkT
-		eth1Data  Eth1DataT
-	)
+	if err := st.SetSlot(0); err != nil {
+		return nil, err
+	}
+
+	var fork ForkT
 	fork = fork.New(
 		genesisVersion,
 		genesisVersion,
 		math.U64(constants.GenesisEpoch),
 	)
-
-	if err := st.SetSlot(0); err != nil {
-		return nil, err
-	}
-
 	if err := st.SetFork(fork); err != nil {
 		return nil, err
 	}
 
 	// Eth1DepositIndex will be set in processDeposit
 
-	if err := st.SetEth1Data(
-		eth1Data.New(
-			common.Root{},
-			0,
-			executionPayloadHeader.GetBlockHash(),
-		)); err != nil {
+	var eth1Data Eth1DataT
+	eth1Data = eth1Data.New(
+		common.Root{},
+		0,
+		execPayloadHeader.GetBlockHash(),
+	)
+	if err := st.SetEth1Data(eth1Data); err != nil {
 		return nil, err
 	}
 
-	// TODO: we need to handle common.Version vs
-	// uint32 better.
-	bodyRoot := blkBody.Empty(version.ToUint32(genesisVersion)).HashTreeRoot()
-	if err := st.SetLatestBlockHeader(
-		blkHeader.New(
-			0,             // slot
-			0,             // proposer index
-			common.Root{}, // parent block root
-			common.Root{}, // state root
-			bodyRoot,
-		)); err != nil {
+	// TODO: we need to handle common.Version vs uint32 better.
+	var blkBody BeaconBlockBodyT
+	blkBody = blkBody.Empty(version.ToUint32(genesisVersion))
+
+	var blkHeader BeaconBlockHeaderT
+	blkHeader = blkHeader.New(
+		0,                      // slot
+		0,                      // proposer index
+		common.Root{},          // parent block root
+		common.Root{},          // state root
+		blkBody.HashTreeRoot(), // body root
+
+	)
+	if err := st.SetLatestBlockHeader(blkHeader); err != nil {
 		return nil, err
 	}
 
 	for i := range sp.cs.EpochsPerHistoricalVector() {
 		if err := st.UpdateRandaoMixAtIndex(
 			i,
-			common.Bytes32(executionPayloadHeader.GetBlockHash()),
+			common.Bytes32(execPayloadHeader.GetBlockHash()),
 		); err != nil {
 			return nil, err
 		}
 	}
 
+	// Process deposits
 	for _, deposit := range deposits {
 		if err := sp.processDeposit(st, deposit); err != nil {
 			return nil, err
 		}
 	}
 
-	// TODO: process activations.
-	validators, err := st.GetValidators()
-	if err != nil {
-		return nil, err
-	}
```

Comment:
nit: just reduced `validators` scope, since this is only used right below to set genesis validators root